src/common/snippets/src/op/subgraph.cpp

```diff
@@ -432,21 +432,20 @@ void snippets::op::Subgraph::initialize_buffer_scratchpad_size() {
         // Propagate to up: in Store. Buffer can have only one Store
         {
             auto parent = buffer->get_input_node_shared_ptr(0);
-            auto idx = buffer->input(0).get_source_output().get_index();
-            // There may be graph with several LoopBegin and LoopEnd between Store/Brgemm and Buffer,
-            // so we should iterate through LoopBase
-            while (ov::is_type<snippets::op::LoopBase>(parent)) {
-                const auto source_output = parent->input_value(idx);
-                parent = source_output.get_node_shared_ptr();
-                idx = source_output.get_index();
-            }
-            if (auto store = ov::as_type_ptr<snippets::op::Store>(parent)) {
-                store->set_offset(offset);
-            } else if (const auto brgemm = ov::as_type_ptr<snippets::op::Brgemm>(parent)) {
-                // Brgemm encapsulates work with loading and storing of data
-                brgemm->set_offset_c(offset);
-            } else {
-                throw ngraph_error("Buffer::set_offset() was called when Buffer didn't have the corresponding Store op for offset propagation");
+            if (!ov::is_type<ngraph::op::v0::Constant>(parent)) {
+                auto idx = buffer->input(0).get_source_output().get_index();
+                while (ov::is_type<snippets::op::LoopBase>(parent)) {
```

Comment:
This approach works nicely in the cases when there are several consecutive LoopEnd's, and it's perfect for now. However, it would throw in the following pattern:
```
EltwiseSomething
Store (connected to Buffer)
LoopEnd
HorizonSomething
LoopEnd
Buffer
```
Let's leave it as is, but keep such cases in mind for the future.